chainconsumer/analysis.py

```diff
@@ -269,11 +270,13 @@ def _get_smoothed_histogram(self, chain, parameter):
             hist = gaussian_filter(hist, smooth, mode=self.parent._gauss_mode)
         kde = chain.config["kde"]
         if kde:
-            kde_xs = np.linspace(edge_centers[0], edge_centers[-1], max(200, int(bins.max())))
-            ys = MegKDE(data, chain.weights, factor=kde).evaluate(kde_xs)
-            area = simps(ys, x=kde_xs)
-            ys = ys / area
-            ys = interp1d(kde_xs, ys, kind="linear")(xs)
+            # kde_xs = np.linspace(edge_centers[0], edge_centers[-1], max(200, int(bins.max())))
+            # ys = MegKDE(data, chain.weights, factor=kde).evaluate(kde_xs)
+            # area = simps(ys, x=kde_xs)
+            # ys = ys / area
+            # ys = interp1d(kde_xs, ys, kind="linear")(xs)
+            ys, kde_xs = fastKDE.pdf(data)
```

Comment:
**Important comment here**

So far the only real issue I can think of is that I don't see how fastKDE supports weighted data. Most samplers that arent emcee or HMC return weighted samples, so its a requirement to support that. I might have missed something and it does support it, let me know if you can find a way for it to do so, because I am coming up blank :s

Reply:
Looking at various options in the fastKDE docstrings (the documentation is rather sparse...), it seems there are no options to include weights yet :/

Reply:
Ive flicked an email to Travis to see what our options are. Hopefully he gets back soon or it might end up floating in my inbox whilst I am away!

Reply:
Update, Travis said a few people have requested the feature, but he doesnt have firm plans to implement it, as it would require some theoretical work first. I'll read his papers and collaborate with him when I am back from break, and hopefully we can get this pulled in then.  :)